### PR TITLE
[Wayland] publish TestServer fixture

### PR DESCRIPTION
**Pull request by AlanGriffiths**
Add the miral::TestServer fixture to the mirtest package

This is a part of what a downstream would need to test their own server.

### CODE REVIEW

**Comment by AlanGriffiths** [comment]
I plan to extract parts of the "wlcs integration" that will be useful downstream in a later PR.

File: tests/mir_test_framework/test_server.cpp [hunk 1]
```diff
@@ -210,13 +169,30 @@ void miral::TestDisplayServer::invoke_tools(std::function<void(WindowManagerTool
...
+void TestDisplayServer::add_server_init(std::function<void(mir::Server&)>&& init)
 {
-    if (auto const wm = window_manager.lock())
-        f(*wm);
-    else
-        BOOST_THROW_EXCEPTION(std::runtime_error{"Server not running"});
+    auto temp = [old_init=init_server, new_init=std::move(init)](mir::Server& server)
+        {
+            old_init(server);
+            new_init(server);
```
Comment (wmww) [inline tests/mir_test_framework/test_server.cpp]: This is a little strange, but I suppose it works just as well as pushing to a vector of functions.

**Comment by AlanGriffiths** [comment]
bors =wmww

**Comment by AlanGriffiths** [comment]
bors r=wmww

**Pull request merged**